data/transactions/logic/eval.go

```diff
@@ -2525,6 +2526,145 @@ func opEd25519verify(cx *EvalContext) {
 	cx.stack = cx.stack[:prev]
 }
 
+// opEcDsaVerify verifies ECDSA signature.
+// It accepts both compressed and uncompressed forms of PK because
+// the underlying secp256k1.VerifySignature decompressed it anyway.
+// Rationale here is the fact compressed PKs look more commonly used in Eth world rather uncompressed.
+func opEcDsaVerify(cx *EvalContext) {
+	ecdsaCurve := EcDsaCurve(cx.program[cx.pc+1])
+	fs, ok := ecDsaCurveSpecByField[ecdsaCurve]
+	if !ok || fs.version > cx.version {
+		cx.err = fmt.Errorf("invalid curve %d", ecdsaCurve)
+		return
+	}
+
+	if fs.field != Secp256k1 {
+		cx.err = fmt.Errorf("unsupported curve %d", fs.field)
+		return
+	}
+
+	last := len(cx.stack) - 1 // index of PK x uncompressed or a full PK compressed
+	prev := last - 1          // index of PK y uncompressed or empty
+	pprev := prev - 1         // index of signature r
+	fourth := pprev - 1       // index of signature s
+	fifth := fourth - 1       // index of data
+
+	pkX := cx.stack[last].Bytes
+	pkY := cx.stack[prev].Bytes
+	sigR := cx.stack[pprev].Bytes
+	sigS := cx.stack[fourth].Bytes
+	msg := cx.stack[fifth].Bytes
+
+	var pubkey []byte
+	// if PK y component is set then interpret it as uncompressed PK
+	if len(pkY) != 0 {
+		x := new(big.Int).SetBytes(pkX)
+		y := new(big.Int).SetBytes(pkY)
+		pubkey = secp256k1.CompressPubkey(x, y)
```

Comment:
Instead of compressing the PK and making `VerifySignature` decompress it, you can encode it as an uncompressed key. I believe that can be done with `secp256k1.S256().Marshal(x, y)`.

Reply:
done